### PR TITLE
chore: type optimization

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | chore: type optimization |
| 🇨🇳 Chinese | chore: type optimization |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/steps/index.tsx [hunk 1]
```diff
@@ -62,7 +66,7 @@ const Steps: StepsType = props => {
   const { xs } = useBreakpoint(responsive);
   const { getPrefixCls, direction: rtlDirection } = React.useContext(ConfigContext);
 
-  const getDirection = React.useCallback(
+  const getDirection = React.useMemo<RcStepsProps['direction']>(
```
Comment (afc163) [inline components/steps/index.tsx:69]: 这个为啥改 useMemo？

Comment (li-jia-nan) [reply components/steps/index.tsx:69]: `useMemo` 返回的是函数运行的结果，`useCallback` 返回的是函数，这里需要的是一个值，没必要用 `useCallback` 并且执行拿到返回结果，这个场景使用 `useMemo` 更加合适一点

Comment (afc163) [reply components/steps/index.tsx:69]: 那这个命名有问题，不应该叫 getDirection

File: components/steps/index.tsx [hunk 2]
```diff
@@ -62,7 +66,7 @@ const Steps: StepsType = props => {
   const { xs } = useBreakpoint(responsive);
   const { getPrefixCls, direction: rtlDirection } = React.useContext(ConfigContext);
 
-  const getDirection = React.useCallback(
+  const _direction = React.useMemo<RcStepsProps['direction']>(
```
Comment (afc163) [inline components/steps/index.tsx:69]:
```suggestion
  const realDirectionValue = React.useMemo<RcStepsProps['direction']>(
```

**Review by afc163 — APPROVED** [review]

**Pull request merged**